includes/components/class-user.php

```diff
@@ -55,18 +55,45 @@ public function __construct( $args = [] ) {
 
 			add_action( 'personal_options_update', [ $this, 'update_profile_fields' ], 100 );
 			add_action( 'edit_user_profile_update', [ $this, 'update_profile_fields' ], 100 );
+
+			// Set user status.
+			add_action( 'admin_init', [ $this, 'set_user_status' ] );
 		} else {
 
 			// Redirect author page.
 			add_action( 'template_redirect', [ $this, 'redirect_author_page' ] );
 
 			// Alter templates.
 			add_filter( 'hivepress/v1/templates/site_footer_block', [ $this, 'alter_site_footer_block' ] );
+
+			// Set user status.
+			add_action( 'init', [ $this, 'set_user_status' ] );
 		}
 
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Updates user online status.
+	 */
+	public function set_user_status() {
+
+		// Get online users.
+		$online_users = (array) get_transient( 'online_users' );
```

Comment:
Please use some prefix e.g. hp_, some plugins may use the same transient name.

Reply:
By the way our Cache API uses transients and adds prefixes etc, you can use

hivepress()->cache->set_cache(...

instead of the WP-level Transient API. Please always prefer the framework API when available, this way we can change the implementation in a single place if WP changes something, or add middleware between the calls and actual WP API if required.